# Survey Analysis: Unravelling Consumer Awareness of Eco-Labels in Kenya: Assessing the Impact of Multiple Schemes and Exploring the KEBS Framework for Enhanced Sustainable Consumption

In this notebook we will examine and test 3 hypotheses to help us assess the impact of multiples ecolabelling schemes on consumers in Kenya

In [11]:
#import the neccesary libaries and data
import pandas as pd
import numpy as np
import seaborn as sns
data = pd.read_csv("data.csv")
data

Timestamp  \
0   9/21/2023 18:44:35   
1   9/21/2023 19:01:12   
2   9/21/2023 19:07:56   
3   9/21/2023 19:17:04   
4   9/21/2023 20:01:56   
..                 ...   
76   9/29/2023 8:58:41   
77   9/29/2023 9:02:53   
78  10/1/2023 14:20:39   
79  10/6/2023 11:55:10   
80  10/9/2023 11:02:53   

   I confirm that I have read the information above and understand that my participation is voluntary.  \
0                                                 Yes                                                    
1                                                 Yes                                                    
2                                                 Yes                                                    
3                                                 Yes                                                    
4                                                 Yes                                                    
..                                                ...                                                    
76                                                Yes                                                    
77                                                Yes                                                    
78                                                Yes                                                    
79                                                Yes                                                    
80                                                Yes                                                    

   Kindly enter your email address to proceed What is your age?  \
0                      susanmuia128@gmail.com             18-24   
1                    c.gichuki@alustudent.com             18-24   
2                      eddiepkhiras@gmail.com             55-64   
3                 kirangakariukilaw@gmail.com             25-34   
4                      negrilson20@gmail.com              35-44   
..                                        ...               ...   
76                      kndirangu44@gmail.com             35-44   
77                       gaatuomar@gmail.com              35-44   
78                      brianomega8@gmail.com             35-44   
79                         mcnjathi@gmail.com             25-34   
80                     r.okoko@alustudent.com             25-34   

   What is your Gender? Which region in Kenya do you currently reside in?  \
0                Female                                           Nairobi   
1                Female                                           Central   
2                  Male                                             Coast   
3                  Male                                           Nairobi   
4                  Male                                            Nyanza   
..                  ...                                               ...   
76                 Male                                             Coast   
77                 Male                                     North Eastern   
78                 Male                                           Nairobi   
79                 Male                                           Nairobi   
80                 Male                                           Nairobi   

    How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                   5                                           
1                                                   5                                           
2                                                   5                                           
3                                                   4                                           
4                                                   4                                           
..                                                ...                                           
76                

# Hypothesis 1
###   Multiple eco-labelling schemes create confusion among consumers in Kenya and fail to promote awareness of eco-friendly products effectively.

**Specific Question for Testing: "How often do you find it challenging to understand the environmental implications of a product based on its eco-label?"**
Let's focus on one specific aspect of the hypothesis: the relationship between encountering multiple eco-labels and the challenge of understanding environmental implications.


*H0:There is no significant relationship between encountering multiple eco-labels and the frequency of finding it challenging to understand the environmental implications of a product based on its eco-label.*
*H1:There is a significant relationship between encountering multiple eco-labels and the frequency of finding it challenging to understand the environmental implications of a product based on its eco-label*


In [12]:
# Chi-squared testing
from scipy.stats import chi2_contingency


#Create a subset of the data from to include only the relevant questions
d1 = data[['Have you ever encountered multiple eco-labels on similar products?','How often do you find it challenging to understand the environmental implications of a product based on its eco-label?']]

# Create a contingency table
contingency_table = pd.crosstab(d1['Have you ever encountered multiple eco-labels on similar products?'], d1['How often do you find it challenging to understand the environmental implications of a product based on its eco-label?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 6.1537404902789525
P-value: 0.10436533940361604
Degrees of freedom: 3
Expected frequencies table:
[[ 9.33333333  2.07407407 20.22222222 10.37037037]
 [ 8.66666667  1.92592593 18.77777778  9.62962963]]


## Results and Interpretation at 95% Confidence
The Chi-squared Statistic: 6.154

The Chi-squared statistic measures the strength of association between the two categorical variables, which in this case are "Encountering Multiple Eco-Labels" and "Challenges Understanding Eco-Labels."
P-value: Approximately 0.104

The p-value indicates the probability of obtaining the observed relationship (or a stronger one) between the variables by random chance, assuming there is no actual association.
Degrees of Freedom: 3

The degrees of freedom represent the number of categories or cells in the contingency table that are free to vary.
Interpretation at a 95% Confidence Level:

At a 95% confidence level, corresponding to a significance level (alpha) of 0.05, we aim to be 95% confident in our conclusions and consider the results statistically significant if the p-value is less than 0.05.

The p-value we obtained is approximately 0.104, which is greater than 0.05. Therefore, there is not enough evidence to reject the null hypothesis at the 95% confidence level.

Interpretation and Implications:

In the context of our hypothesis, which posits that "Multiple eco-labelling schemes create confusion among consumers in Kenya and fail to promote awareness of eco-friendly products effectively," the results of this analysis suggest that there is no statistically significant relationship between encountering multiple eco-labels and the frequency of finding it challenging to understand the environmental implications based on eco-labels. In other words, this specific analysis does not provide strong evidence to conclude that encountering multiple eco-labels leads to a higher frequency of challenges in understanding eco-labels.

It's important to note that this result is specific to the data and analysis conducted. Several factors may influence consumer perceptions and eco-label effectiveness, and this analysis is just one piece of the puzzle. Further research and exploration of these variables may provide a more comprehensive understanding of consumer behaviors and eco-labeling schemes in Kenya.

Additionally, it's worth considering that the non-significant result does not necessarily imply the absence of any effect. It could be influenced by the specific sample size, the choice of survey questions, or other external factors. Therefore, this result should be considered in conjunction with other research findings and real-world observations.

By adding context and discussing the implications, your research report will provide a more comprehensive understanding of the results and their significance within the broader research landscape.


# Hypothesis 2
### How much trust do you place in products that bear the KEBS label when it comes to quality and safety?

#### Specific Questions for Testing:

**How much trust do you place in products that bear the KEBS label when it comes to quality and safety?**

**Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?**

Null Hypothesis (H0): There is no significant relationship between respondents' trust in products with the KEBS label and their likelihood to intentionally purchase products with the KEBS label.

Alternative Hypothesis (H1): There is a significant relationship between respondents' trust in products with the KEBS label and their likelihood to intentionally purchase products with the KEBS label

In [13]:
d2 = data[['How much trust do you place in products that bear the KEBS label when it comes to quality and safety?','Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?']]
d2

How much trust do you place in products that bear the KEBS label when it comes to quality and safety?  \
0                                                Some                                                      
1                                        A grest deal                                                      
2                                        A grest deal                                                      
3                                              Little                                                      
4                                              Little                                                      
..                                                ...                                                      
76                                           Somewhat                                                      
77                                       A great deal                                                      
78                                       A great deal                                                      
79                                             Little                                                      
80                                       A great deal                                                      

   Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?  
0                                                 Yes                                                                    
1                                                 Yes                                                                    
2                                                 Yes                                                                    
3                                                  No                                                                    
4                                                  No                                                                    
..                                                ...                                                                    
76                                                 No                                                                    
77                                                Yes                                                                    
78                                                Yes                                                                    
79                                                Yes                                                                    
80                                                Yes                                                                    

[81 rows x 2 columns]

In [14]:
from scipy.stats import chi2_contingency

# Create a contingency table
contingency_table = pd.crosstab(d2['How much trust do you place in products that bear the KEBS label when it comes to quality and safety?'], d2['Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 32.185034534033
P-value: 5.460451677135931e-06
Degrees of freedom: 5
Expected frequencies table:
[[ 5.97530864 16.02469136]
 [ 0.54320988  1.45679012]
 [ 5.97530864 16.02469136]
 [ 2.17283951  5.82716049]
 [ 0.27160494  0.72839506]
 [ 7.0617284  18.9382716 ]]


At 95% Confidence Level:

With a 95% confidence level, we would typically use a significance level (alpha) of 0.05. Let's interpret the results at this level:

Chi-squared Statistic: The Chi-squared statistic is approximately 32.185.

This statistic measures the strength of the association between trust in products with the KEBS label and the intention to purchase products with the KEBS label.
P-value: The p-value is approximately 5.46e-06, which is very close to zero.

A p-value of 5.46e-06 is much less than 0.05 (the significance level for a 95% confidence level). This indicates a highly significant relationship.
Interpretation at 95% Confidence:

At a 95% confidence level, the results provide strong evidence to reject the null hypothesis. This suggests that there is a significant relationship between respondents' trust in products with the KEBS label and their intention to purchase products with the KEBS label. In the context of our hypothesis, this would indicate that trust in KEBS-labeled products is related to the intention to purchase such products.

At 90% Confidence Level:

With a 90% confidence level, we would typically use a significance level (alpha) of 0.10. Let's interpret the results at this level:

Chi-squared Statistic: The Chi-squared statistic remains approximately 32.185.

P-value: The p-value remains approximately 5.46e-06.

Interpretation at 90% Confidence:

At a 90% confidence level, the results still provide very strong evidence to reject the null hypothesis. This indicates a highly significant relationship, similar to what was observed at the 95% confidence level.

Summary:

In both cases, at both the 95% and 90% confidence levels, the results strongly support the alternative hypothesis, suggesting a significant relationship between trust in products with the KEBS label and the intention to purchase products with the KEBS label. This finding is consistent with the hypothesis that trust in KEBS-labeled products affects the intention to purchase them.

# Hypothesis 3: To what extent do you associate the KEBS label with government oversight and regulation?


Specific Questions for Testing:

"To what extent do you associate the KEBS label with government oversight and regulation?"

"Do you believe that products with the KEBS label are more reliable and safer than those without it?"

Null Hypothesis (H0): There is no significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label.

Alternative Hypothesis (H1): There is a significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label.

In [15]:
#Select the relevant columns
d3 = data[['To what extent do you associate the KEBS label with government oversight and regulation?','Do you believe that products with the KEBS label are more reliable and safer than those without it?']]
d3

To what extent do you associate the KEBS label with government oversight and regulation?  \
0                                             Neutral                                         
1                                            Strongly                                         
2                                            Strongly                                         
3                                             Neutral                                         
4                                            Somewhat                                         
..                                                ...                                         
76                                           Somewhat                                         
77                                           Strongly                                         
78                                           Strongly                                         
79                                           Not very                                         
80                                           Strongly                                         

   Do you believe that products with the KEBS label are more reliable and safer than those without it?  
0                                                 Yes                                                   
1                                                 Yes                                                   
2                                                 Yes                                                   
3                                                  No                                                   
4                                                  No                                                   
..                                                ...                                                   
76                                                Yes                                                   
77                                                Yes                                                   
78                                                Yes                                                   
79                                                 No                                                   
80                                                Yes                                                   

[81 rows x 2 columns]

In [16]:

# Create a contingency table
contingency_table = pd.crosstab(d3['To what extent do you associate the KEBS label with government oversight and regulation?'], d3['Do you believe that products with the KEBS label are more reliable and safer than those without it?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 30.351315789473684
P-value: 4.151045276326365e-06
Degrees of freedom: 4
Expected frequencies table:
[[ 4.66666667  9.33333333]
 [ 0.66666667  1.33333333]
 [ 3.33333333  6.66666667]
 [ 6.33333333 12.66666667]
 [12.         24.        ]]


**Results and Intepration**

At 95% Confidence Level:

With a 95% confidence level, we would typically use a significance level (alpha) of 0.05. Let's interpret the results at this level:

Chi-squared Statistic: The Chi-squared statistic is approximately 30.351.

P-value: The p-value is approximately 4.15e-06, which is very close to zero.

Interpretation at 95% Confidence:

At a 95% confidence level, the results provide strong evidence to reject the null hypothesis. This suggests that there is a significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label.

At 90% Confidence Level:

With a 90% confidence level, we would typically use a significance level (alpha) of 0.10. Let's interpret the results at this level:

Chi-squared Statistic: The Chi-squared statistic remains approximately 30.351.

P-value: The p-value remains approximately 4.15e-06.

Interpretation at 90% Confidence:

At a 90% confidence level, the results still provide very strong evidence to reject the null hypothesis. This indicates a highly significant relationship, similar to what was observed at the 95% confidence level.

Summary:

In both cases, at both the 95% and 90% confidence levels, the results strongly support the alternative hypothesis. This finding suggests a significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label. The evidence is consistent with the hypothesis that such association influences perceptions of product reliability and safety

# Justification for Using the Chi-squared Test:

The Chi-squared test is suitable for your hypotheses for the following reasons:

Categorical Data: The survey responses are largely categorical in nature. The Chi-squared test is designed for the analysis of categorical data and can assess associations between categorical variables.

Independence Testing: The Chi-squared test assesses the independence of two categorical variables. It's particularly well-suited for testing whether there's a significant relationship between the categorical variables in your hypotheses.

Hypothesis-Driven: Your hypotheses involve relationships between categorical variables, making the Chi-squared test a hypothesis-driven choice.

Chi-squared Test of Independence: For your hypotheses related to associations and relationships among categorical variables, the Chi-squared test of independence is a commonly used and straightforward statistical test.

While other tests can be valuable for different types of data and hypotheses, the Chi-squared test aligns with the nature of your survey questions and the specific relationships you want to investigat